test/fw/ptl/utils/pbs_testsuite.py

```diff
@@ -248,6 +248,30 @@ def wrap_obj(obj):
     return wrap_obj
 
 
+def testparams(**kwargs):
+    """
+    Decorator to set or modify test specific parameters
+    """
+    def decorated(function):
+        function.__doc__ += "Test Params:" + "\n\t"
+        for key, value in kwargs.iteritems():
+            function.__doc__ += str(key) + ' : ' + str(value) + '\n\t'
+
+        def wrapper(self, *args):
+            for key, value in kwargs.iteritems():
+                keyname = str(self.__class__.__name__) + "." + key
+                if keyname not in self.conf.keys():
+                    self.conf[keyname] = value
+                    self.testconf[keyname] = value
+                else:
+                    self.testconf[keyname] = self.conf[keyname]
```

Comment:
At this point, you know the default value of keyname.
In theory, you can use `type(value)`  to tell you the type, and then cast `self.conf[keyname]` to that value.
The code could look like this:
```Python
t = type(value)
if t is bool: # this would require a special function, as casting any string with len > 1 to bool will return True
    self.testconf[keyname] = strasbool(self.conf[keyname]) 
else:
    self.testconf[keyname] = t(self.conf[keyname])
```

Reply:
Thanks @vstumpf , I have addressed your comments, Please review.